# Ćwiczenie 1

In [1]:
import pandas as pd
import numpy as np
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)


0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


In [2]:
# 1a
import pandas as pd

pensje = pd.Series({
    'Anna': 5500,
    'Jan': 7200,
    'Ewa': 4800,
    'Marek': 9100,
    'Kasia': 6300
})

print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


In [3]:
#1b
# 1. Kto zarabia najwięcej
print(pensje.idxmax())

# 2. Średnia pensja
print(pensje.mean())

# 3. Osoby powyżej średniej
print(pensje[pensje > pensje.mean()].index)

# 4. Ile osób poniżej 6000
print((pensje < 6000).sum())

Marek
6580.0
Index(['Jan', 'Marek'], dtype='str')
2


In [4]:
#1c
# 1. Podwyżka 10%
podwyzka = pensje * 1.10

# 2. Premia 500 zł
premia = pensje + 500

# 3. Nowa pensja (podwyżka + premia)
nowe_pensje = (pensje * 1.10) + 500

# 4. Ranking od najwyższej
ranking = nowe_pensje.sort_values(ascending=False)

print(ranking)

Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


## Ćwiczenie 2 


In [5]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})


In [ ]:
# 2a
# 1. Kształt tabeli
print(sklep.shape)

# 2. Nazwy kolumn
print(sklep.columns)

# 3. Typy danych
print(sklep.dtypes)

# 4. Pierwsze 3 wiersze
print(sklep.head(3))

(5, 4)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


In [ ]:
# 2b
# 1. Wartość magazynu
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']

# 2. Cena z VAT
sklep['cena_z_vat'] = sklep['cena'] * 1.23

# 3. Wyświetlenie całości
print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [ ]:
# 2c
import pandas as pd

url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

# 1. Kształt datasetu
print(tips.shape)

# 2. Ostatnie 5 wierszy
print(tips.tail())

# 3. 3 losowe wiersze
print(tips.sample(3))

# 4. Typy danych
print(tips.dtypes)

(244, 7)
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2
     total_bill  tip   sex smoker  day    time  size
163       13.81  2.0  Male     No  Sun  Dinner     2
187       30.46  2.0  Male    Yes  Sun  Dinner     5
28        21.70  4.3  Male     No  Sat  Dinner     2
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


## Ćwiczenie 3 

In [ ]:
# Dane 
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

In [ ]:
# 3a
# 1. head(10)
print(penguins.head(10))

# 2. shape
print(penguins.shape)

# 3. info()
penguins.info()

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

In [ ]:
# 3b
# 1. describe()
print(penguins.describe())

# 2. describe dla wszystkiego
print(penguins.describe(include='all'))

# 3. Mediana body_mass_g
print(penguins['body_mass_g'].median())

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   

In [ ]:
# 3c
# 1. Gatunki
print(penguins['species'].value_counts())

# 2. Wyspy
print(penguins['island'].value_counts())

# 3. Płeć
print(penguins['sex'].value_counts())

# 4. Procent Adelie
print(penguins['species'].value_counts(normalize=True))

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64
sex
MALE      168
FEMALE    165
Name: count, dtype: int64
species
Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674
Name: proportion, dtype: float64


In [ ]:
# 3d
# 1. Braki w każdej kolumnie
print(penguins.isna().sum())

# 2. % braków w sex
braki_sex = penguins['sex'].isna().sum()
procent = braki_sex / len(penguins) * 100
print(procent)

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64
3.1976744186046515


## Wnioski 3e
-Dataset zawiera 344 pingwiny opisane w 7 kolumnach, obejmujących cechy fizyczne oraz informacje o środowisku.

-W danych występują 3 gatunki pingwinów, z czego najliczniejszy jest gatunek Adelie (około 44%), następnie Gentoo i Chinstrap.

-Pingwiny zamieszkują 3 wyspy, a najwięcej osobników znajduje się na wyspie Biscoe (około 49% wszystkich obserwacji).

-Średnia masa ciała pingwinów wynosi około 4200 g, natomiast mediana to około 4050 g, co sugeruje lekko prawostronny rozkład danych.

-W datasetcie występują brakujące dane, szczególnie w kolumnie sex (około 3% braków) oraz pojedyncze braki w innych kolumnach numerycznych.

-Rozkład płci jest w miarę zrównoważony, choć liczba samców jest nieznacznie większa niż samic.

# Ćwiczenie 4  

In [ ]:
# Dane
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

In [ ]:
#4
# 1. Średni napiwek (%)
print(tips['tip_pct'].mean())

# 2. Dzień z największą liczbą rachunków
print(tips['day'].value_counts())

# 3. Rachunki od palaczy
print(tips['smoker'].value_counts())

# 4. Describe dla tip + mediana
print(tips['tip'].describe())
print("Mediana napiwku:", tips['tip'].median())

# 5. % rachunków Dinner
print(tips['time'].value_counts(normalize=True))

# 6. Konkretnie Dinner:
print(tips['time'].value_counts(normalize=True)['Dinner'])

# 7. Liczba unikalnych rozmiarów grup
print(tips['size'].nunique())

16.08155737704918
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64
smoker
No     151
Yes     93
Name: count, dtype: int64
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
Mediana napiwku: 2.9
time
Dinner    0.721311
Lunch     0.278689
Name: proportion, dtype: float64
0.7213114754098361
6
